test/modules/TestResultCoordinator/report/CountingReportGenerator.cs

```diff
@@ -71,137 +59,94 @@ sealed class CountingReportGenerator : ITestResultReportGenerator
         /// <returns>Test Result Report.</returns>
         public async Task<ITestResultReport> CreateReportAsync()
         {
-            Logger.LogInformation($"Start to generate report by {nameof(CountingReportGenerator)} for Sources [{this.expectedSource}] and [{this.actualSource}]");
-
-            long lastLoadedKeyFromExpectStore = -1;
-            long lastLoadedKeyFromActualStore = -1;
             TestOperationResult lastLoadedResult = default(TestOperationResult);
-
-            var expectQueue = new Queue<TestOperationResult>();
-            var actualQueue = new Queue<TestOperationResult>();
             ulong totalExpectCount = 0;
             ulong totalMatchCount = 0;
             ulong totalDuplicateResultCount = 0;
             List<TestOperationResult> unmatchedResults = new List<TestOperationResult>();
-            ulong duplicatesFound = 0;
 
-            (lastLoadedKeyFromExpectStore, _, _) = await this.LoadBatchIntoQueueAsync(this.expectedSource, this.expectedStore, lastLoadedKeyFromExpectStore, expectQueue);
-            (lastLoadedKeyFromActualStore, lastLoadedResult, duplicatesFound) = await this.LoadBatchIntoQueueAsync(this.actualSource, this.actualStore, lastLoadedKeyFromActualStore, actualQueue, lastLoadedResult);
-            totalDuplicateResultCount += duplicatesFound;
+            bool hasExpectedResults = await this.ExpectedResults.MoveNextAsync();
+            TestOperationResult expectedResult = this.ExpectedResults.Current;
+            bool hasActualResults = await this.ActualResults.MoveNextAsync();
+            TestOperationResult actualResult = this.ActualResults.Current;
 
-            while (expectQueue.Count > 0 && actualQueue.Count > 0)
+            while (hasExpectedResults && hasActualResults)
```

Comment:
is it possible to refactor like below?
```
bool verificationLoopCompleted = false;

while (!verificationLoopCompleted)
{
    bool hasExpectedResult = await this.ExpectedTestResultSource.MoveNextAsync();
    bool hasActualResult = await this.ActualTestResultSource.MoveNextAsync();
    verificationLoopCompleted = !hasExpectedResults || !hasActualResults;
    
    if (verificationLoopCompleted)
    {
        continue;
    }

    while (hasActualResult && this.TestResultComparer.Matches(lastLoadedResult, this.ActualTestResultSource.Current))
    {
        totalDuplicateResultCount++;
        lastLoadedResult = this.ActualTestResultSource.Current;
        hasActualResults = await this.ActualResults.MoveNextAsync();

        if (!hasActualResults)
        {
            verificationLoopCompleted = true;
            continue;
        }
    }

    totalExpectCount++;
    
    if .....
    ..... 
}
```

Reply:
I think I like it better with the MoveNext in the begginig than checking again in the loop if it should continue